OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -1960,9 +2090,7 @@ static void sendPurchases(JSONArray purchases, boolean newAsExisting, OneSignalR
             jsonBody.put("existing", true);
          jsonBody.put("purchases", purchases);
 
-         OneSignalRestClient.post("players/" + getUserId() + "/on_purchase", jsonBody, responseHandler);
-         if (getEmailId() != null)
-            OneSignalRestClient.post("players/" + getEmailId() + "/on_purchase", jsonBody, null);
+         OneSignalStateSynchronizer.sendPurchases(jsonBody, responseHandler);
       } catch (Throwable t) {
          Log(LOG_LEVEL.ERROR, "Failed to generate JSON for sendPurchases.", t);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.